libr/core/cmd_search.c

```diff
@@ -1549,6 +1550,134 @@ static void do_esil_search(RCore *core, struct search_parameters *param, const c
 	r_cons_clear_line (1);
 }
 
+static void do_syscall_search(RCore *core, struct search_parameters *param) {
+	RSearch *search = core->search;
+	ut64 at;
+	ut64 curpc;
+	ut8 *buf;
+	ut8 *arr;
+	RAnalOp aop;
+	int i,j,inslen, ret, bsize = R_MIN (64, core->blocksize);
+	int kwidx = core->search->n_kws;
+	int count = 0;	
+	int nb_opcodes = 8;
+	ut64 old_offset = core->offset;
+	int nbytes = 0;
+	RAnalEsil *esil = core->anal->esil;
+	RIOMap* map;
+	RListIter *iter; 
+	int stacksize = r_config_get_i (core->config, "esil.stack.depth");
+	int bits   = r_config_get_i (core->config, "asm.bits");
+	int iotrap = r_config_get_i (core->config, "esil.iotrap");
+	
+	if (!(esil = r_anal_esil_new (stacksize, iotrap))) {
+		return;
+	}	
+	buf = malloc (bsize);	
+	arr = malloc (bsize);
+	if (!buf || !arr) {
+		eprintf ("Cannot allocate %d bytes\n", bsize);
```

Comment:
Memleak

Reply:
I can't understand how this leads to Memleak , pls help me in this !!

Reply:
free(buf);
free (arr);
to fix the memleak